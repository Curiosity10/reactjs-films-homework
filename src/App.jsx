import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { SearchResultsPageContainer } from './pages/SearchResultsPage';
import { MovieDetailsPageContainer } from './pages/MovieDetailsPage';
import store from './modules/store';
import ErrorBoundary from './components/ErrorBoundary';
import './styles/App.scss';

const App = () => (
  <Provider store={store}>
    <ErrorBoundary>
      <BrowserRouter>
        <Switch>
          <Route exact path="/" component={SearchResultsPageContainer} />
          <Route
            path="/movie/:id"
            render={({ match }) => {
              const id = match.params.id.slice(1);
              return <MovieDetailsPageContainer id={id} />;
            }}
          />
        </Switch>
      </BrowserRouter>
    </ErrorBoundary>
  </Provider>
);

export default App;
