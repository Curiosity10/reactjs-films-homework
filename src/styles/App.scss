@import "fonts.scss";

* {
  margin: 0;
  padding: 0;
  font-family: "Helvetica Neue", Arial, Helvetica, sans-serif;
  font-size: 16px;
}
