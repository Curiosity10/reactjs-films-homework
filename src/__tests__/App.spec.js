import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';

describe('App component renders correctly', () => {
  it('App renders correctly with main page', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<App />);
    expect(result).toMatchSnapshot();
  });

  it('App renders correctly with movie details page', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(
      <MemoryRouter initialEntries={['/movie/123']}>
        <App />
      </MemoryRouter>,
    );
    expect(result).toMatchSnapshot();
  });
});
