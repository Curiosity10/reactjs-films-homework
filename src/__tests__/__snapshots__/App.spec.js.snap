// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly App renders correctly with main page 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ErrorBoundary>
    <BrowserRouter>
      <Switch>
        <Route
          path="/movie/:id"
          render={[Function]}
        />
        <Route
          render={[Function]}
        />
      </Switch>
    </BrowserRouter>
  </ErrorBoundary>
</Provider>
`;

exports[`App component renders correctly App renders correctly with movie details page 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "y46ah7",
          "pathname": "/movie/123",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "y46ah7",
        "pathname": "/movie/123",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <App />
</Router>
`;
